calico_node/allocate_ipip_addr/allocate_ipip_addr.go

```diff
@@ -0,0 +1,112 @@
+package allocateIPIPAddr
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"debug/pe"
+)
+
+func allocateIPIPAddr(client *client.Client, string nodename) {
+	meta := api.IPPoolMetadata{}
+	ipPoolList, err := client.IPPools().List(meta)
+	if err != nil {
+		log.Warnf("error retrieving pools: %s", err)
+	}
+    ipv4Pools := getIPv4Pools(ipPoolList.Items)
+    enabledPools := getIPIPEnabledPools(ipv4Pools)
+    if len(enabledPools) > 0 {
+        ensureHostTunnelAddress(ipv4Pools, enabledPools, client, nodename)
+    } else {
+        removeHostTunnelAddr(client, nodename)
+    }
+}
+
+func ensureHostTunnelAddress(ipv4Pools, enabledPools []api.IPPool, client *client.Client, string nodename){
+    cnf = client.Config()
+    ipAddr, err := cnf.GetNodeIPIPTunnelAddress(nodename)
+    if err != nil {
+        panic(fmt.Sprintf("Could not retrieve IPIP tunnel address: %s", err))
```

Comment:
Panics output stack trace.  I don't think we want to be doing that.  Better to just output a sensible error message and exit with a non-zero RC.